ggshield/sca/output/schemas.py

```diff
@@ -0,0 +1,78 @@
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Literal, Optional, cast
+
+import marshmallow_dataclass
+from pygitguardian.models import Base, BaseSchema, FromDictMixin
+
+
+@dataclass
+class SCAVulnerability(Base, FromDictMixin):
+    severity: str
+    summary: str
+    cve_ids: list[str] = field(default_factory=list)
+    created_at: Optional[datetime] = None
+    fixed_version: Optional[str] = None
+
+
+SCAVulnerability.SCHEMA = cast(
+    BaseSchema,
+    marshmallow_dataclass.class_schema(SCAVulnerability, base_schema=BaseSchema)(),
+)
+
+SCADependencyType = Literal["direct", "transitive"]
+
+
+@dataclass
+class SCAVulnerablePackageVersion(Base, FromDictMixin):
+    package_full_name: str
+    version: str
+    ecosystem: str
+    dependency_type: Optional[SCADependencyType] = None
```

Comment:
When can we have `dependency_type` set to None?

Reply:
We parse dependency files in `mescal` to declare a dependency as `direct` or `transitive`, but sometimes we do not know. We then set it to `None`.